packages/py-moose-lib/moose_lib/dmv2/view.py

```
@@ -5,39 +5,54 @@
-        teardown = [f"DROP VIEW IF EXISTS {name}"]
-        super().__init__(
-            name, setup, teardown, pulls_data_from=base_tables, metadata=metadata
-        )
+        self.name = name
+        self.select_sql = select_statement
+        self.source_tables = [t.name for t in base_tables]
+        self.metadata = metadata
+        self.source_file = get_source_file_from_stack()
+
+        if self.name in _custom_views:
+            raise ValueError(f"View with name {self.name} already exists")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)